Pal/src/host/Linux-SGX/pal_linux.h

```diff
@@ -158,15 +158,15 @@ static inline __attribute__((always_inline))
 char * __hex2str(void * hex, int size)
 {
     static char * ch = "0123456789abcdef";
-    char * str = __alloca(size * 2);
+    char * str = __alloca(size * 2 + 1);
 
     for (int i = 0 ; i < size ; i++) {
         unsigned char h = ((unsigned char *) hex)[i];
         str[i * 2] = ch[h / 16];
         str[i * 2 + 1] = ch[h % 16];
     }
 
-    str[size * 2 - 1] = 0;
+    str[size * 2] = 0;
     return str;
```

Comment:
Huh, this function returns a buffer from alloca...

Reply:
I agree that this is not good; this may be ok because it is inlined, but still seems that the buffer should be alloca'ed in the parent function to be safe.  This function is not widely used (just looks like a few debugging output cases), but I would still prefer to fix this while we are "in the pool".